tests/rptest/scale_tests/large_messages_test.py

```diff
@@ -0,0 +1,488 @@
+
+        # Set message processing limit
+        messages_per_second_per_producer = 0
+
+        # Start redpanda
+        self.redpanda.start()
+
+        # Do create topics stage
+        self._create_topics()
+
+        # Do the healthcheck on RP
+        # to make sure that all topics are settle down and have their leader
+        self._wait_until_cluster_healthy()
+
+        # Calculate some values
+        total_bytes = self.n_topics * self.message_count * self.message_size
+        total_mb = total_bytes / 1024 / 1024
+        if messages_per_second_per_producer:
+            # By default, we expect ~20MB per sec
+            # I.e. 20 topics, 1 messages per topic, 1MB message size
+            running_time_sec = \
+                self.message_count // messages_per_second_per_producer
+            expected_min_throughput = total_bytes / \
+                (self.message_count // messages_per_second_per_producer)
+        else:
+            # If there is no limit on throughput, use 2 min value
+            running_time_sec = 120
+            # and use scale parameter for max throughput
+            expected_min_throughput = self.scale.expect_single_bandwidth
+
+        # Lower throughput expectation by 10% to account for network errors
+        # and swarm node slow start/end
+        expected_min_throughput *= 0.9
+        expected_throughput_mb = expected_min_throughput / 1024 / 1024
+        self.logger.info(f"Total data: {total_mb:.2f}MB ({total_bytes}b), "
+                         f"Expected throughput >{expected_throughput_mb}MB/s")
+
+        # # Run swarm consumers
+        _group = "large_messages_group"
+        swarm_consumers = self._run_consumers_with_constant_rate(
+            self.n_topics, self.topic_prefixes, _group)
+
+        # Run swarm producers
+        swarm_producers = self._run_producers_with_constant_rate(
+            self.n_topics, self.topic_prefixes,
+            messages_per_second_per_producer)
+
+        # Wait for all messages to be produced
+        self.logger.info("Measuring bandwidth")
+        # Measure bandwidth each 2 seconds
+        # if no new bytes received by RP, check swarm and exit
+        # if at least one finished
+        bandwidth_in = []
+        bandwidth_out = []
+        backoff_interval = 2
+        seconds_spent = 0
+        last_read, last_sent = self._get_rw_metrics()
+        interval_start_sec = time.time()
+        while seconds_spent < running_time_sec:
+            read, sent = self._get_rw_metrics()
+            # Since we are doing some other calculations,
+            # Use timing to determine elapsed interval
+            interval_end_sec = time.time()
+            elapsed_sec = interval_end_sec - interval_start_sec
+            # Calculate ingress BW
+            bytes_per_sec_in = (read - last_read) / elapsed_sec
+            mb_per_sec_in = bytes_per_sec_in / 1024 / 1024
+            # Calculate egress BW
+            bytes_per_sec_out = (sent - last_sent) / 2
```

Comment:
What is the halving for?  Should we involve `elapsed_sec` here?

Reply:
Yes, it is a mistype. elapsed_sec is correct here.